rclcpp/src/rclcpp/node.cpp

```diff
@@ -359,7 +359,7 @@ Node::declare_parameter(
 void
 Node::undeclare_parameter(const std::string & name)
 {
-  this->node_parameters_->undeclare_parameter(name);
+  this->node_parameters_->force_undeclare_parameter(name);
```

Comment:
Should this also be passing a boolean `false`? The local node can use `force_undeclare_parameter` to ensure it's undeclared, but having a non-forced one would catch programmer errors.

Reply:
I'm not sure I understand.
Are you suggesting to expose both the forced and non-forced methods as part of the node interface? such that users can choose which one to use?
(and then internally use a flag)

Reply:
Yes. I wouldn't say it's a hard requirement, but I think it would be better for users to default to using the unforced one unless they know what they're doing.

Reply:
240012a should address this.
I also added the changes to the lifecycle nodes, since I forgot before.